src/rviz_visual_tools.cpp

```diff
@@ -1566,6 +1566,32 @@ bool RvizVisualTools::publishPath(const EigenSTL::vector_Vector3d &path, const c
   return triggerInternalBatchPublishAndDisable();
 }
 
+bool RvizVisualTools::publishPath(const std::vector<Eigen::Vector3d> &path, const std::vector<colors> &colors, const double radius,
+                                  const std::string &ns)
+{
+  if (path.size() < 2)
+  {
+    ROS_WARN_STREAM_NAMED(name_, "Skipping path because " << path.size() << " points passed in.");
+    return true;
+  }
+
+  if (path.size() != colors.size())
+  {
+    ROS_ERROR_STREAM_NAMED(name_, "Skipping path because " << path.size() << " different from " << colors.size() << ".");
+    return false;
+  }
+
+  // Batch publish, unless it is already enabled by user
+  enableInternalBatchPublishing(true);
+
+  // Create the cylinders
+  for (std::size_t i = 1; i < path.size(); ++i)
+    publishCylinder(path[i - 1], path[i], colors[i], radius, ns);
```

Comment:
I was already using the `LINE_STRIP` before.
We should add it in this library, it works great; the problems with the viewpoint have been mostly solved in the last RViz update (where they also introduced anti-aliasing).

I'm using this to simulate wire deposition (3D printing) so cylinders are very useful because setting the radius allows to get a visualisation close the the reality!

I wanted to add spheres at each intersection of cylinders but the rendering of the `SPHERE_LIST` is really weird (it looks like a 2D sprite).